src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1311,6 +1311,73 @@ void CodeGen::genAvxOrAvx2Intrinsic(GenTreeHWIntrinsic* node)
             break;
         }
 
+        case NI_AVX_SetAllVector256:
+        {
+            assert(op1 != nullptr);
+            assert(op2 == nullptr);
+            op1Reg = op1->gtRegNum;
+            if (varTypeIsIntegral(baseType))
+            {
+                // If the argument is a integer, it needs to be moved into a XMM register
+                regNumber tmpXMM = node->ExtractTempReg();
+                if (varTypeIsLong(baseType))
+                {
+                    emit->emitIns_R_R(INS_mov_i2xmm, emitTypeSize(baseType), tmpXMM, op1Reg);
+                }
+                else
+                {
+                    emit->emitIns_R_R(INS_mov_i2xmm, emitTypeSize(TYP_SIMD16), tmpXMM, op1Reg);
+                }
+                op1Reg = tmpXMM;
+            }
+
+            if (compiler->compSupports(InstructionSet_AVX2))
+            {
+                emit->emitIns_R_R(ins, emitTypeSize(TYP_SIMD32), targetReg, op1Reg);
+            }
+            else
+            {
+                // duplicate the scalar argument to XMM register
+                switch (baseType)
+                {
+                    case TYP_FLOAT:
+                        emit->emitIns_SIMD_R_R_I(INS_vpermilps, emitTypeSize(TYP_SIMD16), op1Reg, op1Reg, 0);
+                        break;
+                    case TYP_DOUBLE:
+                        emit->emitIns_R_R(INS_movddup, emitTypeSize(TYP_SIMD16), op1Reg, op1Reg);
+                        break;
+                    case TYP_BYTE:
+                    case TYP_UBYTE:
+                    {
+                        regNumber tmpZeroReg = node->GetSingleTempReg();
+                        emit->emitIns_R_R(INS_pxor, emitTypeSize(TYP_SIMD16), tmpZeroReg, tmpZeroReg);
+                        emit->emitIns_SIMD_R_R_R(INS_pshufb, emitTypeSize(TYP_SIMD16), op1Reg, op1Reg, tmpZeroReg);
+                        break;
+                    }
+                    case TYP_SHORT:
+                    case TYP_USHORT:
+                        emit->emitIns_SIMD_R_R_I(INS_pshuflw, emitTypeSize(TYP_SIMD16), op1Reg, op1Reg, 0);
```

Comment:
Why `pshuflw` instead of `unpcklwd`?

Reply:
I just followed Clang codegen. They have the same performance on most architectures.

Reply:
Right, but the latter produces smaller code, iirc

Reply:
We can investigate it later.